File: 0two-sum-ii-input-array-is-sorted/0two-sum-ii-input-array-is-sorted.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int n = numbers.Length;
        int l = 0;
        int r = n-1;
        while(l<r){
            int sum = numbers[l] + numbers[r];
            if(sum== target) return new int[2]{l+1,r+1};
            if(sum<target) l+=1;
            else r-=1;
        }
        return new int[2];
    }
}

File: 0two-sum-ii-input-array-is-sorted/0two-sum-ii-input-array-is-sorted.rb
def two_sum(numbers, target)
    left = 0
    right = numbers.length - 1
    while left < right
        curr_sum = numbers[left] + numbers[right]
        if curr_sum == target
            return [left + 1, right + 1]
        elsif curr_sum < target
            left += 1
        else
            right -= 1
        end
    end
    []
end